Content.Shared/Interaction/SharedInteractionSystem.cs

```diff
@@ -1062,7 +1062,6 @@ public bool InteractDoAfter(EntityUid user, EntityUid used, EntityUid? target, E
             if (target == null)
                 return false;
 
-            DebugTools.Assert(!IsDeleted(user) && !IsDeleted(used) && !IsDeleted(target));
```

Comment:
This catching stuff means it's doing its job so the source of it should be fixed instead.

Reply:
Is it tough? For example, in the case of clicking a tray with a carrot, the assert fails becaused `!IsDeleted(used)` returns true, which I suppose is because the carrot gets deleted in the handling of `AfterInteractEvent`. In my understanding, this is not a case that should make that assert fail. Maybe something like this can be added just before it:
```C#
if (IsDeleted(used))
{
    return;
}
```

Reply:
For that specific assert, if the event was handled by some system and had some effect, the event should have been marked as handled. AFAIK the only way that this specific assert could fail is if something didn't properly handle the interaction.